src/SFML/Window/CMakeLists.txt

```diff
@@ -288,6 +288,11 @@ target_link_libraries(sfml-window PUBLIC SFML::System)
 # glad sources
 target_include_directories(sfml-window SYSTEM PRIVATE "${PROJECT_SOURCE_DIR}/extlibs/headers/glad/include")
 
+# When static linking on macOS, we need to add this flag for objective C to work
+if ((NOT BUILD_SHARED_LIBS) AND SFML_OS_MACOSX)
+    target_link_libraries(sfml-window PRIVATE -ObjC)
```

Comment:
While we're here, this should probably be using [`target_link_options`](https://cmake.org/cmake/help/latest/command/target_link_options.html) instead. Based on the [Apple docs](https://developer.apple.com/library/archive/qa/qa1490/_index.html), this just needs to be a linker flag.

Reply:
I tried out `target_link_options` and got the same runtime failure as before. Good idea to link to the docs though. I added that.